Source/Core/Core/HW/DVD/AMMediaboard.h

```diff
@@ -0,0 +1,217 @@
+// Copyright 2017 Dolphin Emulator Project
+// SPDX-License-Identifier: GPL-2.0-or-later
+
+#pragma once
+
+#include <memory>
+#include <optional>
+#include <string>
+#include <vector>
+
+#include "Common/CommonTypes.h"
+
+class PointerWrap;
+
+namespace Core
+{
+class System;
+}
+
+namespace File
+{
+class IOFile;
+}
+
+enum GameType
+{
+  FZeroAX = 1,
+  FZeroAXMonster,
+  MarioKartGP,
+  MarioKartGP2,
+  VirtuaStriker3,
+  VirtuaStriker4,
+  GekitouProYakyuu,
+  KeyOfAvalon,
+  FirmwareUpdate
+};
+enum MediaType
+{
+  GDROM = 1,
+  NAND,
+};
+enum MediaBoardType
+{
+  NANDMaskBoardHDD = 0,
+  NANDMaskBoardMask = 1,
+  NANDMaskBoardNAND = 2,
+  DIMMBoardType3 = 4,
+};
+enum MediaBoardStatus
+{
+  Initializing = 0,
+  CheckingNetwork = 1,
+  SystemDisc = 2,
+  TestingGameProgram = 3,
+  LoadingGameProgram = 4,
+  LoadedGameProgram = 5,
+  Error = 6
+};
+enum InquiryType
+{
+  Version1 = 0x21484100,
+  Version2 = 0x29484100
+};
+
+#define SocketCheck(x) (x <= 0x3F ? x : 0)
+
+namespace AMMediaboard
+{
+enum class AMMBCommand : u16
+{
+  Unknown_000 = 0x000,
+  GetDIMMSize = 0x001,
+
+  Inquiry = 0x12,
+  Read = 0xa8,
+  Write = 0xaa,
+  Execute = 0xab,
```

Comment:
I don't think these 4 belong in `AMMBCommand`. `AMMBCommand` seems to correspond to `media_buffer_in_16[1]`, while these 4 commands correspond to `DICMDBUF[0]`. These should be in their own enum (maybe called `AMMBDICommand`, though that's a lot of letters in a row).